packages/cli-repl/src/mongosh-repl.ts

```diff
@@ -646,17 +647,39 @@ class MongoshNodeRepl implements EvaluationListener {
    * @param result A ShellResult (or similar) object.
    * @returns The pretty-printed version of the input.
    */
-  formatShellResult(result: { type: null | string, printable: any }): string {
-    return this.formatOutput({ type: result.type, value: result.printable });
+  formatShellResult(
+    result: { type: null | string; printable: any },
+    extraFormatOptions: Partial<FormatOptions> = {}
+  ): string {
+    return this.formatOutput(
+      { type: result.type, value: result.printable },
+      extraFormatOptions
+    );
   }
 
   /**
    * Called when print(), console.log() etc. are called from the shell.
    *
    * @param values A list of values to be printed.
    */
-  onPrint(values: ShellResult[]): void {
-    const joined = values.map((value) => this.formatShellResult(value)).join(' ');
+  onPrint(values: ShellResult[], type: 'print' | 'printjson'): void {
+    const extraOptions: Partial<FormatOptions> | undefined =
+      // MONGOSH-955: when `printjson()` is called in mongosh, we will try to
+      // replicate the format of the old shell: disable colors, start every
+      // object on the new line, and set all the collapse options threshold to
+      // infinity
+      type === 'printjson'
+        ? {
+          colors: false,
```

Comment:
I don’t think the syntax highlighting was really brought up as a problem, right? I’m not sure if there’s really much point in disabling it, we only use colors when printing to outputs that support it anyway.

Reply:
Yeah, I can remove this, I was thinking that if people want us to add support for this because they are using this programmatically, maybe it's safer to remove any unexpected symbols from the output, which ansi color codes might be, but I might be just over cautious here

Reply:
Yeah, I think our color support detection code is pretty solid and we haven’t ever received a complaint about it, so I think it’s safe to leave the color support in the default mode here.

Also, to be clear, this PR currently wouldn’t give users programmatically usable output. We did discuss the option of making `printjson()` print actual (E)JSON, but ultimately felt that that would also not really be what users want, despite the name and everything.

Reply:
Oh, yeah, totally! I guess I meant usable in users sense 😄 I agree that otherwise this method is pretty confusing and should probably be avoided